libyul/optimiser/UnusedFunctionsCommon.h

```diff
@@ -48,7 +48,7 @@ std::vector<T> filter(std::vector<T> const& _vec, std::vector<bool> const& _mask
 
 /// Returns true if applying UnusedFunctionParameterPruner is not helpful or redundant because the
 /// inliner will be able to handle it anyway.
-bool tooSimpleToBePruned(FunctionDefinition const& _f)
+inline bool tooSimpleToBePruned(FunctionDefinition const& _f)
```

Comment:
Don't we need to inline the template too?

Reply:
And shouldn't that template deserve to be moved to `libsolutil`?

Reply:
templates are - at least in the sense of symbol collisions - inline by default (you almost always have to define them in headers anyways, to be able to instantiate them on demand - I guess the compiler is usually clever enough not to *actually* just blindly inline, but will still deduplicate implementations within reason).

Reply:
> And shouldn't that template deserve to be moved to `libsolutil`?

Isn't it rather specific for that?

Reply:
> templates are - at least in the sense of symbol collisions - inline by default 

Yep, but it seems the style in the project is to mark them explicitly, at least in the files I looked.

Reply:
*Really*? I was actually confused, why you'd suggest it, but that explains that... looking for example at ``libsolutil/CommonData.h``, it looks pretty inconsistent to me, though, with maybe half of the templates using ``inline``, half of them not using it...
Personally, I'd vote for not doing it, ever - and e.g. adding ``Do not use "inline". The only exception are very short, free, non-template helper functions shared among the code base that are defined in common headers files.`` to the coding style... or sth like that...

Reply:
I would say "use inline only when required to avoid symbol collisions".